### ISSUE TITLE
[Housekeeping] Type safety for <cookbook-showcase-properties> columns

### ISSUE THREAD

**Issue by jkaltoft** [kirbydesign/designsystem#1348]
**Please add a short description of your enhancement request**

Interface `ShowCasePropertyColumns` was added to [apps/cookbook/src/app/shared/showcase-properties/showcase-property.ts](https://github.com/kirbydesign/designsystem/blob/master/apps/cookbook/src/app/shared/showcase-properties/showcase-property.ts) in another recent issue for a specific case. However, applying it to other similar cases were out of scope for that issue.

**Describe the solution you'd like**

I'd like to apply it everywhere we use `<cookbook-showcase-properties>` (ShowcasePropertiesComponent) with columns other than the default to enforce type safety.

There are currently ~13 cases using non-default column headers.

_In addition, `ShowCasePropertyColumns` should be renamed to `ShowcasePropertyColumns` (with a lowercase c)._

**Describe alternatives you've considered**

NONE

**Additional context**

_Example (Radio showcase component):_

```js
eventColumns = {
    Name: 'Name',
    Description: 'Description',
    Type: 'Signature',
};
```

should be changed to:

```js
eventColumns: ShowcasePropertyColumns = {
    Name: 'Name',
    Description: 'Description',
    Type: 'Signature',
};
```

which prevents adding object keys that the `ShowcasePropertiesComponent` does not use.

![image](https://user-images.githubusercontent.com/150451/108852675-32320980-75e6-11eb-80f2-7d99ca8b1a83.png)

![image](https://user-images.githubusercontent.com/150451/108852769-4bd35100-75e6-11eb-8bdc-91833ff3a777.png)

### Kick Off:

- [ ] Ensure the enhancement is `UX refined` and aligned with UX 
_The component and/or enhancement should be published and available in the [Kirby Styleguide on Zeplin](https://zpl.io/258pXGj)_
- [x] Ensure the enhancement has been `Tech refined` with @kirbydesign/kirby-guild and this issue is updated with a clear implementation description  
_This issue should be in the [Ready to do](https://github.com/kirbydesign/designsystem/projects/1#column-4590936) column of the [Kirby kan-ban board](https://github.com/kirbydesign/designsystem/projects/1) before starting implementation)_
- [x] Assign yourself to this issue and move it to the [In progress](https://github.com/kirbydesign/designsystem/projects/1#column-4590937) column of the [Kirby kan-ban board](https://github.com/kirbydesign/designsystem/projects/1)

### Code:

- [x] Create Feature Branch from [master branch](https://github.com/kirbydesign/designsystem/tree/master)
- [x] Create a draft implementation and push to Github
- [x] Ask a member of @kirbydesign/kirby-guild for a WIP review by creating a draft Pull Request
- [ ] Implement unit tests
- [ ] Update Cookbook Examples and Showcase, i.e. see [Radio](https://cookbook.kirby.design/home/showcase/radio)  
_Also remember to add any relevant new API documentation_

### Review:

- UX review:
  - [ ] Ensure implementation is correct in relation to the UX design and the [Kirby Styleguide on Zeplin](https://zpl.io/258pXGj)
  - [ ] With UX agree on the version of the implementation
- Code review:
  - [x] Open a pull request (or mark the existing draft PR as `Ready for review`) and ask @kirbydesign/kirby-guild for a review  
_Remember to add `closes #issueno` to the description of the PR._
  - [ ] Once approved, merge feature branch/PR to master
- [ ] Ask a member of @kirbydesign/kirby-guild to add a link to component showcase from Kirby Component Status and update the version number

:tada: Celebrate

**Comment by jkaltoft** [comment]
During the initial code review scope for this issue was expanded. The latest PR is a refactoring issue where different showcase components are used for properties, events and methods respectively.
